### PR TITLE
Feature/h2

### PR DESCRIPTION
**Pull request by vamshi-ra**
Added extendability to H2

### CODE REVIEW

**Review by raxkumar-wedaa — COMMENTED** [review]
Review OK!

File: src/utils/jsonToJdl.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ exports.createJdlFromJson = async (fileName, metadata, req, res) => {
         // throw error response
         if (applicationErrorList.length > 0) {
             applicationError[i] = applicationErrorList;
-            console.err(applicationError);
+            console.info(applicationError);
```
Comment (rohith-wedaa) [inline src/utils/jsonToJdl.js]: console.error()

File: src/utils/jsonToJdl.js [hunk 2]
```diff
@@ -52,13 +52,13 @@ exports.createJdlFromJson = async (fileName, metadata, req, res) => {
...
         if (applications[i].prodDatabaseType !== undefined && !databaseTypes.includes(applications[i].prodDatabaseType)) {
-            applicationErrorList.push('Unknow Database Type, database must be among the following: ' + databaseTypes);
+            applicationErrorList.push('Unknown Database Type, database must be among the following: ' + databaseTypes);
         }
         if (applications[i].serverPort === undefined || applications[i].serverPort === '') {
             applicationErrorList.push('Server Port cannot be empty');
         }
         if (
-            applications[i].prodDatabaseType !== undefined &&
+            applications[i].prodDatabaseType !== undefined && applications[i].prodDatabaseType !== 'h2Memory' &&
```
Comment (rohith-wedaa) [inline src/utils/jsonToJdl.js]: if possible positive checks

File: src/utils/jsonToJdl.js [hunk 3]
```diff
@@ -119,6 +119,12 @@ exports.createJdlFromJson = async (fileName, metadata, req, res) => {
         ) {
             databaseType = 'sql';
         }
+        if (
+            applications[i].prodDatabaseType !== undefined &&
+            applications[i].prodDatabaseType.toLowerCase() == 'h2memory'
+        ) {
+            databaseType = 'h2Memory';
+        }
```
Comment (rohith-wedaa) [inline src/utils/jsonToJdl.js:124]: combine conditions for database

File: src/utils/jsonToJdl.js [hunk 4]
```diff
@@ -153,14 +158,16 @@ application {
...
+                    i
+                ].prodDatabaseType.toLowerCase()}\n        prodDatabaseType ${applications[i].prodDatabaseType.toLowerCase()}`
                 : ''
-        }
-        ${databaseType !== 'no' ? `databasePort ${applications[i].databasePort}` : ''}
+            }
+        ${(databaseType === 'h2Memory') ? `devDatabaseType h2Memory\n`
+                : ''}
+
+        ${databaseType !== 'no' && databaseType !== 'h2Memory' ? `databasePort ${applications[i].databasePort}` : ''}
```
Comment (rohith-wedaa) [inline src/utils/jsonToJdl.js:168]: merge with L161

databaseType === 'sql'

**Review by rohith-wedaa — CHANGES_REQUESTED** [review]

**Review by rohith-wedaa — APPROVED** [review]

**Pull request merged**